pandas/tools/merge.py

```diff
@@ -1626,7 +1626,11 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
                 clean_objs.append(v)
             objs = clean_objs
             name = getattr(keys, 'name', None)
-            keys = Index(clean_keys, name=name)
+            # GH 14848
```

Comment:
hmm

maybe this should be handled in the _get_grouper logic then

where you first figure out what the name actually is